File: app/src/main/java/com/apicloud/pkg/sdk/component/MainComponent.java
package com.apicloud.pkg.sdk.component;

import com.apicloud.pkg.sdk.mvp.ui.activity.MainActivity;   

import com.apicloud.pkg.sdk.module.ActivityMainBindingModule;

import dagger.Component;
@Component(dependencies = AppComponent.class,modules = {ActivityMainBindingModule.class})
public interface MainComponent {
    void inject(MainActivity activity);
}

File: app/src/main/java/com/apicloud/pkg/sdk/component/MainComponent.kt
package com.apicloud.pkg.sdk.component

import com.apicloud.pkg.sdk.mvp.ui.activity.MainActivity

//import com.apicloud.pkg.sdk.module.ActivityMainBindingModule

import dagger.Component

@Component(dependencies = [(AppComponent::class)])
interface MainComponent {
    fun inject(activity: MainActivity)
}